src/sentry/workflow_engine/processors/detector.py

```diff
@@ -20,17 +20,16 @@ def process_detectors(
         detector_group_keys = set()
 
         for result in detector_results:
-            if result.state_update_data:
-                if result.state_update_data.group_key in detector_group_keys:
-                    # This shouldn't happen - log an error and continue on, but we should investigate this.
-                    logger.error(
-                        "Duplicate detector state group keys found",
-                        extra={
-                            "detector_id": detector.id,
-                            "group_key": result.state_update_data.group_key,
-                        },
-                    )
-                detector_group_keys.add(result.state_update_data.group_key)
+            if result.group_key in detector_group_keys:
+                # This shouldn't happen - log an error and continue on, but we should investigate this.
```

Comment:
Errors already go to Sentry and should alert us. Either way, I think that in a later pr I change the data model so that this is actually impossible and I removed this code